### PR TITLE
lint against deprecated flow types

### PR DESCRIPTION
**Pull request by kevinbarabash**
I thought I'd try out some of the lint rules that flow has to offer.  This commit originally also included sketchy null checks as well but it turns out those are easier to fix if you have null coalescing which was easy to add but fixing them was still a bit of a pain so I scaled this back to just fix up deprecated types, namely `*`, `$Supertype`, and `$Subtype`.  I also decided to add flow types for enzyme which brought out batch coverage up from about 85% to 95%.

### CODE REVIEW

**Comment by jeresig** [comment]
Deploy preview for *wonder-blocks* ready!

Built with commit 17b927f1b9b9176849dcd428706322347734be4c

https://deploy-preview-468--wonder-blocks.netlify.com

File: packages/wonder-blocks-core/util/add-style.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ import {processStyleList} from "./util.js";
 import type {StyleType} from "./types.js";
 
 // TODO(kevinb): have an a version which uses exact object types
-export default function addStyle<T: React.AbstractComponent<*> | string>(
+export default function addStyle<T: React.AbstractComponent<any> | string>(
```
Comment (jandrade) [inline packages/wonder-blocks-core/util/add-style.js]: Is it strictly necessary to use `any` here? what would happen if you use `React.Node` instead?

File: packages/wonder-blocks-modal/util/maybe-get-portal-mounted-modal-host-element.test.js [hunk 2]
```diff
@@ -30,10 +30,12 @@ describe("maybeGetPortalMountedModalHostElement", () => {
             </div>
         );
         const wrapper = mount(nodes);
-        const candidateElement = wrapper.find("button")[0];
+        const candidateElement = wrapper.find("button").at(0);
```
Comment (jandrade) [inline packages/wonder-blocks-modal/util/maybe-get-portal-mounted-modal-host-element.test.js]: 🎉 this is way better!

File: packages/wonder-blocks-core/util/add-style.js (hunk 1 above)
Comment (kevinbarabash) [reply packages/wonder-blocks-core/util/add-style.js]: `React.Node` isn't a component.

Comment (kevinbarabash) [reply packages/wonder-blocks-core/util/add-style.js]: I wonder if this should be `mixed` and not `any`.

Comment (kevinbarabash) [reply packages/wonder-blocks-core/util/add-style.js]: That didn't work.  I need to figure out how to make this be a subtype of `{}`.  In the meantime though `any` isn't any worse than `*`.

**Pull request merged**